Dockerfile

```diff
@@ -1,24 +1,38 @@
-FROM alpine AS builder
-COPY . /open-adr-ven-library
-WORKDIR /open-adr-ven-library/build/debug
-RUN apk add --no-cache \
-    g++ \
-    cmake \
-    make \
-    xerces-c-dev \
-    curl-dev \
-    tzdata # for UT DateTimeConverterTest.ConvertToTime_t
-RUN cmake ../.. -DTEST=1 && make -j$(nproc)
-
-FROM alpine
-WORKDIR /open-adr-ven-library/build/debug
-RUN apk add --no-cache \
-    libstdc++ \
-    libcurl \
-    xerces-c \
-    tzdata # for UT DateTimeConverterTest.ConvertToTime_t
-COPY --from=builder /open-adr-ven-library/build/debug/liboadr.so .
-COPY --from=builder /open-adr-ven-library/build/debug/liboadrsd.so .
-COPY --from=builder /open-adr-ven-library/build/debug/samplevenmgr .
-COPY --from=builder /open-adr-ven-library/build/debug/testoadr .
-COPY --from=builder /open-adr-ven-library/build/debug/xml xml
+#syntax=docker/dockerfile:1.2
+
+ARG baseimage=ubuntu:20.04
+
+######################################
+# Builder
+######################################
+FROM ${baseimage} AS builder
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN set -ex \
+    &&  echo 'APT::Install-Recommends "false";' > /etc/apt/apt.conf.d/00-no-recommends \
+    &&  echo 'APT::Install-Suggests "false";'   > /etc/apt/apt.conf.d/00-no-suggestions \
+    &&  apt-get update \
+    &&  apt-get install --yes       \
+            ccache                  \
+            cmake                   \
+            g++                     \
+            gcovr                   \
+            libcurl4-gnutls-dev     \
+            libxerces-c-dev         \
+            libpthread-stubs0-dev   \
```

Comment:
why do we need `libpthread-stubs0-dev`?

Reply:
https://github.com/AnalyticsFire/OpenADR-VEN-Library/blob/master/CMakeLists.txt#L68

`samplevenmgr` requires `pthread` library.

Reply:
Ah, right! Usually it is already installed on system but not sure if that's the case in docker.